cdap-test/src/main/java/co/cask/cdap/test/TestManager.java

```diff
@@ -45,6 +50,37 @@ ApplicationManager deployApplication(Id.Namespace namespace,
                                        Class<? extends Application> applicationClz, File... bundleEmbeddedJars);
 
   /**
+   * Deploys an {@link ApplicationTemplate}. Only supported in unit tests.
```

Comment:
everything here has a local implementation for unit tests and a remote implementation that just hits rest endpoints and could be used for tests against a distributed instance.